src/GafferML/TensorToMesh.cpp

```diff
@@ -0,0 +1,177 @@
+//////////////////////////////////////////////////////////////////////////
+//
+//  Copyright (c) 2014, Lucien Fostier All rights reserved.
+//
+//  Redistribution and use in source and binary forms, with or without
+//  modification, are permitted provided that the following conditions are
+//  met:
+//
+//      * Redistributions of source code must retain the above
+//        copyright notice, this list of conditions and the following
+//        disclaimer.
+//
+//      * Redistributions in binary form must reproduce the above
+//        copyright notice, this list of conditions and the following
+//        disclaimer in the documentation and/or other materials provided with
+//        the distribution.
+//
+//      * Neither the name of John Haddon nor the names of
+//        any other contributors to this software may be used to endorse or
+//        promote products derived from this software without specific prior
+//        written permission.
+//
+//  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS
+//  IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+//  THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+//  PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+//  CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+//  EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+//  PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+//  PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+//  LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+//  NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+//  SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+//////////////////////////////////////////////////////////////////////////
+
+#include "GafferML/TensorToMesh.h"
+
+#include "IECoreScene/MeshPrimitive.h"
+
+
+using namespace std;
+using namespace Imath;
+using namespace IECore;
+using namespace IECoreScene;
+using namespace Gaffer;
+using namespace GafferML;
+
+GAFFER_NODE_DEFINE_TYPE( TensorToMesh );
+
+size_t TensorToMesh::g_firstPlugIndex = 0;
+
+TensorToMesh::TensorToMesh( const std::string &name )
+	:	ObjectSource( name, "tensorMesh" )
+{
+
+	storeIndexOfNextChild( g_firstPlugIndex );
+
+	addChild( new TensorPlug( "vertices" ) );
+	addChild( new TensorPlug( "faces" ) );
```

Comment:
I wonder if these are the right names? In Gaffer terminology, `vertices` would probably be `p` or `position` and `faces` would be `vertexIds`. Perhaps you've chosen the names to line up with the models you anticipate feeding this with though?

Reply:
Yes I did name them after the SMPL model I was testing with.
Here is an optional commit that renames everything to match Gaffer's convention:
406a2be932187d074572767cd2088fbaa85e1be2